state/machine.go

```diff
@@ -302,6 +306,52 @@ func (m *Machine) KeepInstance() (bool, error) {
 	return instData.KeepInstance, nil
 }
 
+// CharmProfiles returns the names of any LXD profiles used by the machine,
+// which were defined in the charm deployed to that machine.
+func (m *Machine) CharmProfiles() ([]string, error) {
+	instData, err := getInstanceData(m.st, m.Id())
+	if errors.IsNotFound(err) {
+		err = errors.NotProvisionedf("machine %v", m.Id())
+	}
+	if err != nil {
+		return nil, err
+	}
+	return instData.CharmProfiles, err
```

Comment:
nit: `s/return instData.CharmProfiles, err/return instData.CharmProfiles, nil/` - we're already returning early for the `err`, so this extra `err` will just be nil anyway. This reduces the ambiguity.

Reply:
no idea why i did that, thanks for the catch!